### PR TITLE
Custom locomotor example implementation

### PR DESCRIPTION
**Pull request by Metadorius**
New locomotor code is `{74FC2B59-C2D3-47D7-9D10-93436A34EBB9}`.

wheeee

- [x] Initial working locomotor impl
- [x] Save/load impl.
  - [x] Tested 
- [x] `IPiggyback` impl.
  - [x] Tested
- [x] Docs?
- [x] Credits/changelog

Reviews in progress are welcome.

### CODE REVIEW

**Comment by Metadorius** [comment]
Testing is welcome, to test IPiggy some sort of locomotion inflictor/infliction stopper must be made. cc @Starkku @MortonPL @chaserli

**Comment by Metadorius** [comment]
Also wouldn't mind a review of @CCHyper @tomsons26 on the COM code.

**Comment by Metadorius** [comment]
Builds now and works w/o issues it seems.

Warheads now have `InflictLocomotor` and `RemoveInflictedLocomotor` boolean tags in order to be able to test IPiggyback. Specify correct `Locomotor` and one of the tags for it to work. Note that due to how it works internally currently it's only usable for the test locomotor.

1. What to do with the locomotor?
2. What to do with the warhead?

Revieeeeeeews and further thoughts please. cc @Starkku @chaserli @MortonPL

File: src/Locomotion/TestLocomotionClass.cpp [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+{
+	if (IsMoving)
+	{
+		CoordStruct coord = DestinationCoord;
+
+		/**
+		 *  Rotate the object around the center coord.
+		 */
+		int radius = Unsorted::LeptonsPerCell*2;
+		coord.X += radius * std::sin(Angle);
```
Comment (Starkku) [inline src/Locomotion/TestLocomotionClass.cpp:54]: Gives a build warning `'+=': conversion from 'double' to 'T', possible loss of data`

File: src/Locomotion/TestLocomotionClass.cpp [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+		coord.Y += radius * std::cos(Angle);
```
Comment (Starkku) [inline src/Locomotion/TestLocomotionClass.cpp:55]: Same as above.

File: src/Locomotion/TestLocomotionClass.cpp [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+	{
+        IPersistStreamPtr piggyAsPersist(this->Piggybacker);
+
+        hr = piggyAsPersist->GetClassID(classid);
+    }
+	else
+	{
+        if (reinterpret_cast<IPiggyback*>(this) == nullptr)
+            return E_FAIL;
+
```
Comment (Starkku) [inline src/Locomotion/TestLocomotionClass.cpp:201]: Something going on with indentation and newlines here.

**Review by Starkku — CHANGES_REQUESTED** [review]
Not familiar enough with locomotor logic or COM stuff to comment much on the technical side but it looks correct to me? Assuming it all works correctly. The Warhead stuff is fine too, assuming the same, could look into expanding it to work with other locomotors in future (well, the ones that support piggybacking at least) and/or integration with AE perhaps.

The Warhead feature could technically warrant docs but since it currently only works for testing purposes, I am not sure.

There's a couple of small things that caught my eye though, other than that I guess it is fine.

**Comment by Belonit** [comment]
What about semantic tag support for new locomotives?

**Review by Starkku — APPROVED** [review]
LGTM.

**Review by MortonPL — APPROVED** [review]
No comments either.

**Pull request merged**